### ISSUE TITLE
KeyError in `replicate_parent_ids` when processing OR_GLM-L2-LCFA_G16_s20201882251400_e20201882252000_c20201882252025.nc

### ISSUE THREAD

**Issue by gerritholl** [deeplycloudy/glmtools#77]
When attempting to process the file `OR_GLM-L2-LCFA_G16_s20201882251400_e20201882252000_c20201882252025.nc` glmtools fails with a `KeyError` in `glmtools.io.traversal:replicate_parent_ids`:

```
$ python ~/checkouts/glmtools/examples/grid/make_GLM_grids.py --fixed_grid --split_events --goes_position east --goes_sector conus --dx=2.0 --dy=2.0 --dt 60 -o 'GLMC/1min/{start_time:%Y/%m/%d/%H}/{dataset_name}' /media/x22264/cache/GLM-file-cache/OR_GLM-L2-LCFA_G16_s20201882251400_e20201882252000_c20201882252025.nc
Traceback (most recent call last):
  File "/home/gholl/checkouts/glmtools/examples/grid/make_GLM_grids.py", line 329, in <module>
    gridder(glm_filenames, start_time, end_time, **grid_kwargs)
  File "/opt/pytroll/pytroll_inst/miniconda3-exp/envs/py38/lib/python3.8/site-packages/glmtools/grid/make_grids.py", line 838, in grid_GLM_flashes
    outputs = list(map(this_proc_each_grid, subgrids))
  File "/opt/pytroll/pytroll_inst/miniconda3-exp/envs/py38/lib/python3.8/site-packages/glmtools/grid/make_grids.py", line 896, in proc_each_grid
    glm = GLMDataset(filename, ellipse_rev=ellipse_rev)
  File "/opt/pytroll/pytroll_inst/miniconda3-exp/envs/py38/lib/python3.8/site-packages/glmtools/io/glm.py", line 211, in __init__
    self.__init_parent_child_data()
  File "/opt/pytroll/pytroll_inst/miniconda3-exp/envs/py38/lib/python3.8/site-packages/glmtools/io/glm.py", line 251, in __init_parent_child_data
    flash_ids = self.replicate_parent_ids('flash_id',
  File "/opt/pytroll/pytroll_inst/miniconda3-exp/envs/py38/lib/python3.8/site-packages/glmtools/io/traversal.py", line 180, in replicate_parent_ids
    e_idx = [grouper.groups[eid] for eid in last_replicated_p_ids]
  File "/opt/pytroll/pytroll_inst/miniconda3-exp/envs/py38/lib/python3.8/site-packages/glmtools/io/traversal.py", line 180, in <listcomp>
    e_idx = [grouper.groups[eid] for eid in last_replicated_p_ids]
KeyError: 950168802
```

I obtained the file triggering the error from AWS S3:

```
aws s3 cp s3://noaa-goes16/GLM-L2-LCFA/2020/188/22/OR_GLM-L2-LCFA_G16_s20201882251400_e20201882252000_c20201882252025.nc . --no-sign-request
```

I'm using glmtools latest master (`v0.5.0-140-geee2691`) and have tried this on two different machines.

Perhaps there is something wrong with the file, but in this case it would be nice if `glmtools` could identify that and gracefully continue, in particular when many files are being processed.

**Comment by deeplycloudy** [comment]
@gerritholl I finally got around to writing a fix for this, which is indeed due to an inconsistent set of flash, group, and event IDs. See the fix in #81, which should do what you want. That PR will probably be merged to master sometime this week. I tested on the file above and it works.

**Comment by deeplycloudy** [comment]
The fix in #81 has been merged into master, and I have tested on the file mentioned above, so I'm closing this issue. Thanks for the report!
